packages/flutter_tools/lib/src/test/flutter_web_platform.dart

```diff
@@ -0,0 +1,359 @@
+// Copyright 2019 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+// ignore_for_file: implementation_imports
+
+import 'dart:async';
+
+import 'package:async/async.dart';
+import 'package:http_multi_server/http_multi_server.dart';
+import 'package:path/path.dart' as p; // ignore: package_path_import
+import 'package:shelf/shelf.dart' as shelf;
+import 'package:shelf/shelf_io.dart' as shelf_io;
+import 'package:shelf_packages_handler/shelf_packages_handler.dart';
+import 'package:shelf_static/shelf_static.dart';
+import 'package:shelf_web_socket/shelf_web_socket.dart';
+import 'package:stream_channel/stream_channel.dart';
+import 'package:test/src/runner/browser/browser_manager.dart';
+import 'package:test/src/runner/browser/default_settings.dart';
+import 'package:test/src/runner/executable_settings.dart';
+import 'package:test_api/src/backend/runtime.dart';
+import 'package:test_api/src/backend/suite_platform.dart';
+import 'package:test_api/src/util/stack_trace_mapper.dart';
+import 'package:test_core/src/runner/configuration.dart';
+import 'package:test_core/src/runner/platform.dart';
+import 'package:test_core/src/runner/runner_suite.dart';
+import 'package:test_core/src/runner/suite.dart';
+import 'package:web_socket_channel/web_socket_channel.dart';
+
+import '../artifacts.dart';
+import '../base/file_system.dart';
+import '../cache.dart';
+import '../convert.dart';
+import '../dart/package_map.dart';
+import '../globals.dart';
+
+class FlutterWebPlatform extends PlatformPlugin {
+  FlutterWebPlatform._(this._server, this._config, this._root) {
+    // Look up the location of the testing resources.
+    final Map<String, Uri> packageMap =
+        PackageMap(fs.path.join(Cache.flutterRoot, 'packages', 'flutter_tools'))
+            .map;
+    testUri = packageMap['test'];
+    final shelf.Cascade cascade = shelf.Cascade()
+        .add(_webSocketHandler.handler)
+        .add(packagesDirHandler())
+        .add(_jsHandler.handler)
+        .add(createStaticHandler(
+          fs.path.join(Cache.flutterRoot, 'packages', 'flutter_tools'),
+          serveFilesOutsidePath: true,
+        ))
+        .add(createStaticHandler(_config.suiteDefaults.precompiledPath,
+            serveFilesOutsidePath: true))
+        .add((shelf.Request request) {
+      if (request.requestedUri.path.contains('require.js')) {
+        final File result = fs.file(fs.path.join(
+          artifacts.getArtifactPath(Artifact.engineDartSdkPath),
+          'lib',
+          'dev_compiler',
+          'amd',
+          'require.js',
```

Comment:
minor nit: would it make sense to have a use a K/V pair, filename  -> expected artifact? And maybe a method that gets the content-type based on the extension?

Reply:
How about some getters and a refactored method? :)

Reply:
👍